lib/plugins/transform/typeAdapter.js

```diff
@@ -120,6 +127,36 @@ module.exports = superclass => class extends superclass {
       .then(() => newValue);
   }
 
+  // object
+  objectValuesMergeAdapter(value) {
+    const newValue = {};
+    return Promise.all(Object.keys(value).map(key => super.transform(value[key])
+      .then(tmpValue => ({
+        tmpValue,
+      }))))
+      .then((xforms) => {
+        xforms.forEach(({tmpValue}) => {
+          if (!_.isEmpty(tmpValue) || _.get(this.options, 'typeAdapter.includeEmpty', true)) {
+            if (_.get(this.options, 'typeAdapter.concatArrays', false)) {
+              _.mergeWith(newValue, tmpValue, (objValue, srcValue) => {
+                if (Array.isArray(objValue)) {
+                  if (_.get(this.options, 'typeAdapter.uniqArrays', false)) {
+                    return _.uniq([...objValue, ...srcValue]);
+                  }
+                  return [...objValue, ...srcValue];
+                }
+                return undefined;
+              });
+            }
+            else {
+              _.merge(newValue, tmpValue);
+            }
+          }
+        });
+      })
+      .then(() => newValue);
```

Comment:
`newValue` should be returned from the previous `then(...)`, this is unnecessary.

Reply:
🍝